runtime/lua/vim/highlight.lua

```diff
@@ -26,21 +26,38 @@ end
 --- Highlight the yanked region
 ---
 --- use from init.vim via
----   au TextYankPost * lua require'vim.highlight'.on_yank()
+---   au TextYankPost * lua vim.highlight.on_yank()
 --- customize highlight group and timeout via
----   au TextYankPost * lua require'vim.highlight'.on_yank("IncSearch", 500)
+---   au TextYankPost * lua vim.highlight.on_yank {higroup="IncSearch", timeout=150}
+--- customize conditions (here: do not highlight a visual selection) via
+---   au TextYankPost * lua vim.highlight.on_yank {on_visual=false}
 ---
--- @param higroup highlight group for yanked region
--- @param timeout time in ms before highlight is cleared
--- @param event event structure
-function highlight.on_yank(higroup, timeout, event)
-  event = event or vim.v.event
+-- @param opts dictionary with options controlling the highlight:
+--              - higroup   highlight group for yanked region (default "IncSearch")
+--              - timeout   time in ms before highlight is cleared (default 150)
+--              - on_macro  highlight when executing macro (default false)
+--              - on_visual highlight when yanking visual selection (default true)
+--              - event     event structure (default vim.v.event)
+function highlight.on_yank(opts)
+  vim.validate {
+    opts = { opts,
+    function(t) if t == nil then return true else return type(t) == 'table' end end,
+    'a table or nil to configure options (see `:h highlight.on_yank`)',
+  }}
+  opts = opts or {}
+  local event = opts.event or vim.v.event
+  local on_macro = opts.on_macro or false
+  local on_visual = (opts.on_visual ~= false)
+
+  if (not on_macro) and vim.fn.reg_executing() ~= '' then return end
   if event.operator ~= 'y' or event.regtype == '' then return end
-  higroup = higroup or "IncSearch"
-  timeout = timeout or 500
+  if (not on_visual) and event.visual then return end
+
+  local higroup = opts.higroup or "IncSearch"
+  local timeout = opts.timeout or 150
 
   local bufnr = api.nvim_get_current_buf()
-  local yank_ns = api.nvim_create_namespace('')
+  local yank_ns = api.nvim_create_namespace('hlyank')
```

Comment:
yeah, this should be fine. Only thing would be we might as well move this out of the function call and just set it as a local variable, so we don't have to call the function every time. NANOSECONDS ARE BEING WASTED.

Reply:
Done!